examples/type_inference/type_inference_visitor_unittest.py

```diff
@@ -0,0 +1,243 @@
+"""Unittest for the type_inference_visitor."""
```

Comment:
This file should be under `test/`. You can create a separate subdirectory called `type_inference`.